### PR TITLE
parser: Add more keywords into UnReservedKeyword list

### PR DESCRIPTION
**Pull request by shenli**
See: https://dev.mysql.com/doc/refman/5.7/en/keywords.html
Unreserved keywords can be used as identifiers.

### CODE REVIEW

**Comment by siddontang** [comment]
year and now are time functions, it is ok not adding in function parser?

**Comment by shenli** [comment]
It is OK.
Identifier can be function name. "YEAR" and "NOW"  are all Identifier, so they can be used as builtin function name.

**Comment by siddontang** [comment]
LGTM

**Comment by shenli** [comment]
Merge with master and fix conflict and fix golint error.
PTAL

File: parser/parser.y [hunk 1]
```diff
@@ -1526,15 +1527,17 @@ IndexType:
...
-// TODO: Add Data Type UnReserved Keywords
 UnReservedKeyword:
-	"AUTO_INCREMENT" | "BEGIN" | "BIT" | "BOOL" | "BOOLEAN" | "CHARSET" | "COLUMN" | "COLUMNS" | "DATE" | "DATETIME"
-|	"ENGINE" | "FULL" | "LOCAL" | "NAMES" | "OFFSET" | "PASSWORD" %prec lower_than_eq | "QUICK" | "ROLLBACK" | "SESSION" | "GLOBAL"
-|	"TABLES"| "TEXT" | "TIME" | "TIMESTAMP" | "TRANSACTION" | "TRUNCATE" | "VALUE" | "WARNINGS" | "YEAR" | "NOW"
-|	"SUBSTRING" | "MODE"
-
+	"AUTO_INCREMENT" | "AFTER" | "BEGIN" | "BIT" | "BOOL" | "BOOLEAN" | "CHARSET" | "COLUMNS" | "COMMIT" 
+|	"DATE" | "DATETIME" | "DEALLOCATE" | "DO" | "END" | "ENGINE" | "ENGINES" | "EXECUTE" | "FIRST" | "FULL" 
+|	"LOCAL" | "NAMES" | "OFFSET" | "PASSWORD" %prec lowerThanEq | "PREPARE" | "QUICK" | "ROLLBACK" | "SESSION" | "SIGNED"
```
Comment (siddontang) [inline parser/parser.y]: should we put PASSWORD in a single line, because it has a precedence.

Comment (shenli) [inline parser/parser.y]: Not necessary. We have "|" to separate different rules. So I think the %prec will not affect other rules.

Comment (siddontang) [inline parser/parser.y]: I know that  %prec will not affect other rules. I just think it seems different from others with %prec.

File: parser/parser.y [hunk 2]
```diff
@@ -425,6 +425,7 @@ import (
 
 	Identifier		"identifier or unreserved keyword"
 	UnReservedKeyword	"MySQL unreserved keywords"
+	NotKeywordToken		"Tokes not mysql keyword but treate as keyword"
```
Comment (qiuyesuifeng) [inline parser/parser.y]: Tokens here?

File: parser/parser_test.go [hunk 3]
```diff
@@ -287,6 +284,23 @@ func (s *testParserSuite) TestParser0(c *C) {
...
+	unreservedKws := []string{
+		"auto_increment", "after", "begin", "bit", "bool", "boolean", "charset", "columns", "commit",
+		"date", "datetime", "deallocate", "do", "end", "engine", "engines", "execute", "first", "full",
+		"local", "names", "offset", "password", "prepare", "quick", "rollback", "session", "signed",
+		"start", "global", "tables", "text", "time", "timestamp", "transaction", "truncate", "unknown",
+		"value", "warnings", "year", "now", "substring", "mode",
+	}
+	for _, kw := range unreservedKws {
+		src := fmt.Sprintf("SELECT %s FROM tbl;", kw)
+		fmt.Printf("%s\n", src)
```
Comment (qiuyesuifeng) [inline parser/parser_test.go]: can this line be removed?

File: parser/parser_test.go [hunk 4]
```diff
@@ -287,6 +284,23 @@ func (s *testParserSuite) TestParser0(c *C) {
...
+		l := NewLexer(src)
+		ok := yyParse(l) == 0
+		c.Assert(ok, IsTrue)
```
Comment (qiuyesuifeng) [inline parser/parser_test.go]: use `c.Assert(yyParse(l), Equals, 0)`

File: parser/parser_test.go (hunk 3 above)
Comment (shenli) [inline parser/parser_test.go]: It is debug log. If the test failed. It will report the failed case.

File: parser/parser.y (hunk 2 above)
Comment (shenli) [inline parser/parser.y]: yes

File: parser/parser_test.go (hunk 3 above)
Comment (coocood) [inline parser/parser_test.go]: You can pass `Commentf(src)` as the last argument in `c.Assert` method to do that.

**Comment by shenli** [comment]
PTAL

**Comment by siddontang** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

**Pull request merged**